packages/core/source/legacy/host.ts

```diff
@@ -0,0 +1,335 @@
+              ] satisfies RemoteMutationRecord);
+            } else {
+              records.push([
+                MUTATION_TYPE_UPDATE_PROPERTY,
+                id,
+                key,
+                value,
+              ] satisfies RemoteMutationRecord);
+            }
+          }
+        }
+
+        mutate(records);
+
+        break;
+      }
+
+      default:
+        throw new Error(`Unsupported action type: ${type}`);
+    }
+  };
+}
+
+function adaptLegacyNodeSerialization(
+  child: LegacyRemoteComponentSerialization | LegacyRemoteTextSerialization,
+  options?: LegacyRemoteChannelOptions,
+): RemoteElementSerialization | RemoteTextSerialization {
+  if (child.kind === LEGACY_KIND_TEXT) {
+    return adaptLegacyTextSerialization(child);
+  } else {
+    return adaptLegacyComponentSerialization(child, options);
+  }
+}
+
+function adaptLegacyTextSerialization({
+  id,
+  text,
+}: LegacyRemoteTextSerialization): RemoteTextSerialization {
+  return {
+    id,
+    type: NODE_TYPE_TEXT,
+    data: text,
+  };
+}
+
+function adaptLegacyComponentSerialization(
+  {id, type, props, children}: LegacyRemoteComponentSerialization,
+  options?: LegacyRemoteChannelOptions,
+): RemoteElementSerialization {
+  const element = options?.elements?.[type] ?? type;
+
+  const [fragments, properties] = adaptFragmentsInProps(props);
+
+  return {
+    id,
+    type: NODE_TYPE_ELEMENT,
+    element,
+    properties,
+    children: [
+      ...children.map((child) => {
+        return adaptLegacyNodeSerialization(child, options);
+      }),
+      ...adaptLegacyFragmentsSerialization(fragments, options),
+    ],
+  };
+}
+
+function adaptFragmentsInProps(
+  props: Record<string, unknown>,
+): [
+  fragments: Record<string, LegacyRemoteFragmentSerialization>,
+  properties: Record<string, unknown>,
+] {
+  const fragments: Record<string, LegacyRemoteFragmentSerialization> = {};
+  const properties: Record<string, unknown> = {};
+
+  for (const [key, value] of Object.entries(props)) {
+    if (isFragment(value)) {
+      fragments[key] = value;
+    } else {
+      properties[key] = value;
+    }
+  }
+
+  return [fragments, properties];
+}
+
+function isFragment(prop: unknown): prop is LegacyRemoteFragmentSerialization {
+  return (
+    prop != null &&
+    typeof prop === 'object' &&
+    'kind' in prop &&
+    prop.kind === LEGACY_KIND_FRAGMENT
+  );
+}
+
+function adaptLegacyFragmentsSerialization(
+  fragments: Record<string, LegacyRemoteFragmentSerialization>,
+  options?: LegacyRemoteChannelOptions,
+): RemoteElementSerialization[] {
+  return Object.entries(fragments).map(([slot, fragment]) => {
+    return adaptLegacyFragmentSerialization(slot, fragment, options);
+  });
+}
+
+function adaptLegacyFragmentSerialization(
+  slot: string,
+  fragment: LegacyRemoteFragmentSerialization,
+  options?: LegacyRemoteChannelOptions,
+): RemoteElementSerialization {
+  return {
+    id: fragment.id,
+    element: 'remote-fragment',
```

Comment:
Can we make this element an option you pass in to the main function, like it is for the [`slotProps.wrapper` option in React/ Preact component wrappers](https://github.com/Shopify/remote-dom/tree/main/packages/preact#:~:text=name%20using%20the%20%60slotProps.wrapper%60%20option)?

Reply:
Added a `slotWrapper` property to `options`. But now I don't think we need it. We already have `elements` on the options. We can probably use it for the wrapper as well. Wdyt?

Reply:
good point